Intersect.Client.Core/Entities/Resource.cs

```diff
@@ -358,45 +442,75 @@ private void CalculateRenderBounds()
             }
         }
 
-        if (Texture == null)
+        if (Texture is not { } texture)
         {
             return;
         }
 
-        _renderBoundsSrc.X = 0;
-        _renderBoundsSrc.Y = 0;
-        if (IsDead && Descriptor.Exhausted.GraphicFromTileset)
-        {
-            _renderBoundsSrc.X = Descriptor.Exhausted.X * Options.Instance.Map.TileWidth;
-            _renderBoundsSrc.Y = Descriptor.Exhausted.Y * Options.Instance.Map.TileHeight;
-            _renderBoundsSrc.Width = (Descriptor.Exhausted.Width + 1) * Options.Instance.Map.TileWidth;
-            _renderBoundsSrc.Height = (Descriptor.Exhausted.Height + 1) * Options.Instance.Map.TileHeight;
-        }
-        else if (!IsDead && Descriptor.Initial.GraphicFromTileset)
+        if (CurrentState is not { } graphicState)
         {
-            _renderBoundsSrc.X = Descriptor.Initial.X * Options.Instance.Map.TileWidth;
-            _renderBoundsSrc.Y = Descriptor.Initial.Y * Options.Instance.Map.TileHeight;
-            _renderBoundsSrc.Width = (Descriptor.Initial.Width + 1) * Options.Instance.Map.TileWidth;
-            _renderBoundsSrc.Height = (Descriptor.Initial.Height + 1) * Options.Instance.Map.TileHeight;
+            return;
         }
-        else
+
+        _renderBoundsSrc.X = 0;
+        _renderBoundsSrc.Y = 0;
+
+        switch(graphicState.TextureType)
         {
-            _renderBoundsSrc.Width = Texture.Width;
-            _renderBoundsSrc.Height = Texture.Height;
+            case ResourceTextureSource.Resource:
+                _renderBoundsSrc.Width = texture.Width;
+                _renderBoundsSrc.Height = texture.Height;
+                break;
+
+            case ResourceTextureSource.Tileset:
+                if (IsDead)
+                {
+                    if (graphicState is { MaximumHealth: 0 } deadGraphic)
+                    {
+                        _renderBoundsSrc = new(
+                            deadGraphic.X * _tileWidth,
+                            deadGraphic.Y * _tileHeight,
+                            (deadGraphic.Width + 1) * _tileWidth,
+                            (deadGraphic.Height + 1) * _tileHeight
+                        );
+                    }
+                    else
+                    {
+                        _renderBoundsSrc = new();
+                    }
+                }
+                else if (graphicState is { MinimumHealth: > 0 } aliveGraphic)
+                {
+                    _renderBoundsSrc = new(
+                        aliveGraphic.X * _tileWidth,
+                        aliveGraphic.Y * _tileHeight,
+                        (aliveGraphic.Width + 1) * _tileWidth,
+                        (aliveGraphic.Height + 1) * _tileHeight
+                    );
+                }
```

Comment:
The logic here is almost the same and I'd prefer it not get duplicated, _and_ it seems suspicious that we don't have the "else" case for alive but we do for dead -- store deadGraphic/aliveGraphic in a local variable, and if the variable is null set the src to empty (use `default` instead of `new()`), and if it isn't use the local variable to calculate the positions